lib/src/dartdoc_options.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+///
+/// dartdoc's dartdoc_options.yaml configuration file follows similar loading
+/// semantics to that of analysis_options.yaml,
+/// [documented here](https://www.dartlang.org/guides/language/analysis-options).
+/// It searches parent directories until it finds an analysis_options.yaml file,
+/// and uses built-in defaults if one is not found.
+///
+library dartdoc.dartdoc_options;
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
```

Comment:
I personally avoid single letter `as` imports

Reply:
Yes, they're not great.  I just obliterated them all, replacing the inconsistent imports of path as path/p/etc with "pathLib".